torch/testing/_internal/common_methods_invocations.py

```diff
@@ -7811,16 +7811,24 @@ def gen_shape_kwargs():
 def sample_inputs_binary_cross_entropy_with_logits(
     op_info, device, dtype, requires_grad, **kwargs
 ):
-    _make_tensor = partial(make_tensor, device=device, dtype=dtype, requires_grad=False)
-    for input, target, base_dict in _generate_sample_inputs_nn_loss(
-        op_info, device, dtype, requires_grad, **kwargs
-    ):
-        weight_lst = [None, random.uniform(-9, 9)]
-        pos_weight_lst = [None, random.uniform(-9, 9)]
-        for weight, pos_weight in product(weight_lst, pos_weight_lst):
-            base_dict["weight"] = _make_tensor(input.shape)
-            base_dict["pos_weight"] = _make_tensor(1, low=0)
-            yield SampleInput(input, args=(target,), kwargs=base_dict)
+    make = partial(make_tensor, device=device, dtype=dtype)
+    make_prob = partial(make, low=0, high=1)
+
+    reductions = ("mean", "sum", "none")
+
+    shapes_and_kwargs = [
+        *[(shape, None) for shape in ((), (1,), (S,), (S, S), (S, S, S))],
+        *[((S, S), dict(reduction=reduction)) for reduction in reductions],
+        *[((S, S), dict(reduction=reduction, weight=make((S, S)))) for reduction in reductions],
```

Comment:
Is weight required to have the same shape as the input? Can it be different?

The docs say weight is "a manual rescaling weight if provided it’s repeated to match input tensor shape" [link](https://pytorch.org/docs/stable/generated/torch.nn.functional.binary_cross_entropy_with_logits.html?highlight=binary_cross_entropy_with_logits#torch.nn.functional.binary_cross_entropy_with_logits) which implies that it can be different, but I'm not sure

Reply:
From playing around it appears that it just has to be broadcast-able with the input shape. Should I add more tests with varying broadcast-able shapes?

Reply:
It would be good to add more tests with a broadcast-able shape.

Reply:
Makes sense, just added